tests/unit/client/oauth2.test.ts

```diff
@@ -244,14 +288,23 @@ describe('OAuth2HttpClient', () => {
       confidentialConfig
     );
 
+    nock(mockParams.host)
+      .post(`${AUTH_BASE}/oauth2/tokens`)
+      .basicAuth({ user: 'clientId', pass: 'secret' })
+      .reply(200, { access_token: mockToken });
+
     const authenticateResult =
-      await confidentialHttpWithAuth.extraAuthMethods.authenticate(mockAuth);
+      await confidentialHttpWithAuth.extraAuthMethods.authenticate(
+        emailAuthData
+      );
+    expect(authenticateResult).toStrictEqual({ accessToken: mockToken });
+
+    // Expect successive requests to be authenticated
     nock(mockParams.host).get('/test').reply(200, '');
 
     const result = await confidentialHttpWithAuth.get('test');
```

Comment:
(minor) Could we randomHexString / uuid or something: `mockToken`, we have `test`, `/test` and `test` and it makes the context a little harder to read 🤔 

Same for clientId and secret, could we make them the same references if they're related objects?

Reply:
💯 Will update